src/Core/Settings.h

```diff
@@ -70,6 +70,7 @@ class IColumn;
     M(UInt64, connections_with_failover_max_tries, DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_MAX_TRIES, "The maximum number of attempts to connect to replicas.", 0) \
     M(UInt64, s3_min_upload_part_size, 512*1024*1024, "The minimum size of part to upload during multipart upload to S3.", 0) \
     M(UInt64, s3_max_single_part_upload_size, 64*1024*1024, "The maximum size of object to upload using singlepart upload to S3.", 0) \
+    M(Int64, s3_max_single_read_retries, 16, "The maximum number of retries during single S3 read.", 0) \
```

Comment:
Why Int64, not UInt64 ?

Reply:
And why it is set to 16 by default? I think it is too many.

Reply:
I wanted to have possibility to have infinite retries, by specifying -1. Will 4 be okay? @nikitamikhaylov

Reply:
@excitoon Infinite retries is the worth thing ever. 4 retries is Ok.

Reply:
+